spec/mongo/cursor_spec.rb

```diff
@@ -33,7 +33,7 @@
       end
 
       it 'does not schedule the finalizer' do
-        expect(ObjectSpace).not_to receive(:define_finalizer)
+        expect(ObjectSpace).not_to receive(:define_finalizer).with(Mongo::Cursor, anything)
```

Comment:
After making the change to have the clients reconnect instead of just being closed after random tests, this was failing occasionally due to `define_finalizer` being called with `Monitor`, so I changed this to only assert that it wasn't called with `Cursor`